### PR TITLE
chore: mock data

### PR DESCRIPTION
**Pull request by hn275**

### CODE REVIEW

File: src/mock/index.ts [hunk 1]
```diff
@@ -0,0 +1,60 @@
...
+ * */
+import * as data from "./data.json";
+
+export const mockFetch = (
+  status: "resolve" | "reject",
+  timeout = 0
+): Promise<any> => {
+  return new Promise((resolve, _) => {
+    setTimeout(() => {
+      setTimeout(() => {
```
Comment (Scott-Kenning) [inline src/mock/index.ts]: Why is there 3 separate `setTimeout`s? and only two of them have the `timeout / 2`?

File: src/mock/index.ts [hunk 2]
```diff
@@ -0,0 +1,60 @@
...
+            new Promise((resolve, reject) => {
+              setTimeout(() => {
+                if (status === "reject") {
+                  reject(new Error("rejected"));
+                }
+
+                resolve(data);
+              }, timeout / 2);
+            }),
+          //
```
Comment (Scott-Kenning) [inline src/mock/index.ts]: Even if it's just mock data that will get deleted soon, we should try and avoid random comments like this to make the code easier for other people to read

File: src/mock/index.ts [hunk 3]
```diff
@@ -0,0 +1,60 @@
...
+): Promise<any> => {
+  return new Promise((resolve, _) => {
+    setTimeout(() => {
+      setTimeout(() => {
+        resolve({
+          //
+          json: () =>
+            new Promise((resolve, reject) => {
+              setTimeout(() => {
+                if (status === "reject") {
```
Comment (Scott-Kenning) [inline src/mock/index.ts]: Standard practice is to handle the success case first, then deal with errors after

**Review by Scott-Kenning — CHANGES_REQUESTED** [review]

File: src/mock/index.ts (hunk 1 above)
Comment (hn275) [reply src/mock/index.ts]: I have no idea why, I didn't even see it until now. Thanks!

I've made the changes. Let me know if it there's anything else

File: src/mock/index.ts [hunk 4]
```diff
@@ -0,0 +1,30 @@
...
+import * as data from "./data.json";
+
+export const mockFetch = (
+  status: "resolve" | "reject",
+  timeout = 0
+): Promise<any> => {
+  return new Promise((resolve, _) => {
+    setTimeout(() => {
+      resolve({
+        //
```
Comment (Scott-Kenning) [inline src/mock/index.ts:15]: Still wondering about the 2 lines that just have empty comments, everything else looks good

Comment (Scott-Kenning) [reply src/mock/index.ts:15]: @hn275

Comment (hn275) [reply src/mock/index.ts:15]: I find having those ticks there to separate callbacks a bit easier to catch my eyes than having to trace the indentations.  I can remove them if you want

Comment (Scott-Kenning) [reply src/mock/index.ts:15]: No need, works for me

**Review by Scott-Kenning — APPROVED** [review]

**Pull request merged**